src/stateProviders/provideFenceConnections.js

```diff
@@ -0,0 +1,92 @@
+import { provideState, update } from 'freactal';
+
+import { getFenceUser } from 'services/fence';
+import { getUserStudyPermission } from 'services/fileAccessControl';
+import { FENCES } from 'common/constants';
+import { omit, flatMap, isEmpty } from 'lodash';
+
+export default provideState({
+  initialState: props => ({
+    fenceConnectionsInitialized: false,
+    fenceConnections: {},
+
+    fenceStudiesInitialized: false,
+    fenceStudies: {},
+  }),
+  computed: {
+    fenceAuthStudies: ({ fenceStudies }) =>
+      !isEmpty(fenceStudies) &&
```

Comment:
if this is false, this function will return `false`, is that ok? is it better to return `[]`? it seems we are checking on `fenceAuthStudies.length` anyway around